pkg/hub/submarineragent/controller.go

```diff
@@ -657,13 +659,17 @@ func newSubmarinerManifestWork(managedCluster *clusterv1.ManagedCluster, config
 	return newManifestWork(SubmarinerCRManifestWorkName, managedCluster.Name, config, submarinerCRFile)
 }
 
-func newOperatorManifestWork(managedCluster *clusterv1.ManagedCluster, config interface{}) (*workv1.ManifestWork, error) {
+func newOperatorManifestWork(managedCluster *clusterv1.ManagedCluster, config interface{}, skipOperatorGroup bool) (*workv1.ManifestWork, error) {
 	files := []string{agentRBACFile}
 	if getClusterProduct(managedCluster) == constants.ProductOCP {
 		files = append(files, sccFiles...)
 	}
 
-	files = append(files, operatorFiles...)
+	if skipOperatorGroup {
+		files = append(files, operatorFiles[1:]...)
```

Comment:
This is rather fragile, if we ever update or re-order `operatorFiles` it’s likely that we won’t remember to change this. Please split up `operatorFiles` and handle the two entries separately — something like

```
const operatorGroupFile = "manifests/operator/submariner-operator-group.yaml"
const operatorSubscriptionFile = "manifests/operator/submariner-operator-subscription.yaml"

// ...

if !skipOperatorGroup {
    files = append(files, operatorGroupFile)
}

files = append(files, operatorSubscriptionFile)
```

Reply:
Funnily enough, that is the way I had done initially. Changed ot this to reduce consts/vars. I was considering adding a note to `operatorFiles` declaration to always keep `submariner-operator-group.yaml` at `[0]`.